client/src/main/kotlin/no/nav/su/se/bakover/client/skatteetaten/SkatteClient.kt

```diff
@@ -37,27 +40,74 @@ class SkatteClient(private val skatteetatenConfig: SkatteetatenConfig) : Skatteo
+                        val mappedFeil = when (it.kode) {
+                            SkattedataFeilrespons.Feilkode.`SSG-007` -> SkatteoppslagFeil.FantIkkePerson
+                            SkattedataFeilrespons.Feilkode.`SSG-008` -> SkatteoppslagFeil.FantIkkeSkattegrunnlagForGittÅr
+                            SkattedataFeilrespons.Feilkode.`SSG-010` -> SkatteoppslagFeil.SkattegrunnlagFinnesIkkeLenger
+                            else -> SkatteoppslagFeil.ApiFeil
+                        }
+
+                        return mappedFeil.left()
+                    }
                 } else {
-                    log.info("Vi fikk hentet summert skattegrunnlag fra skatteetaten.")
-                    return objectMapper.readValue(response.body(), SamletSkattegrunnlag::class.java).right()
+                    return objectMapper.readValue(response.body(), SamletSkattegrunnlag::class.java)
+                        .toDomain()
+                        .tapLeft {
+                            log.error("Feil skjedde under mapping av data fra skatteetaten.")
+                            sikkerLogg.error("Feil skjedde under mapping av data fra skatteetaten.", it)
+                        }
+                        .mapLeft { SkatteoppslagFeil.MappingFeil }
                 }
             }
         }.getOrHandle {
+            if (it is JsonMappingException || it is JsonProcessingException) {
+                log.error("Feilet under deserializering i henting av data fra skatteetaten. Melding: ${it.message}")
+                sikkerLogg.error("Feilet under deserializering i henting av data fra skatteetaten.", it)
+
+                return SkatteoppslagFeil.DeserializeringFeil.left()
+            }
+
             log.warn("Fikk en exception ${it.message} i henting av data fra skatteetaten.", it)
             sikkerLogg.warn(
                 "Fikk en exception ${it.message} i henting av data fra skatteetaten. " +
-                    "Request $getRequest er forespørselen mot skatteetaten som feilet."
+                    "Request $getRequest er forespørselen mot skatteetaten som feilet.",
             )
+
             return SkatteoppslagFeil.Nettverksfeil(it).left()
         }
     }
 }
+
+private data class SkattedataFeilrespons(val kode: Feilkode, val melding: String, val korrelasjonsid: String) {
+    /*
+    * Docs: https://skatteetaten.github.io/datasamarbeid-api-dokumentasjon/reference_summertskattegrunnlag.html
+    */
+    enum class Feilkode(val httpKode: Int, val beskrivelse: String) {
+        `SSG-001`(500, "Uventet feil på tjenesten"),
+        `SSG-002`(500, "Uventet feil i et bakenforliggende system"),
+        `SSG-003`(404, "Ukjent url benyttet"),
+        `SSG-004`(401, "Feil i forbindelse med autentisering"),
+        `SSG-005`(403, "Feil i forbindelse med autorisering"),
+        `SSG-006`(400, "Feil i forbindelse med validering av inputdata"),
+        `SSG-007`(404, "Ikke treff på oppgitt personidentifikator"),
```

Comment:
Vi trenger ikke dra inn ordet personidentifikator i vårt domene, hvis vi bare sender de fnr. Kan bare bruke fnr. Jeg mener jeg så noe i service/domain/route (tror jeg).

Reply:
Dette er egentligen deras beskrivelse på de feilkodene, så har bara kopiert in de for å få litt mer oversikt i vår app. Men ja, kunde ha bytt ut ordet personidentifikator mot fnr f.eks.